docs/path-resolution.rst

```diff
@@ -0,0 +1,454 @@
+lets you `import files from HTTP, IPFS and Swarm URLs or refer directly to packages in NPM registry
+<https://remix-ide.readthedocs.io/en/latest/import.html>`_.
+
+.. note::
+
+    Host Filesystem Loader's file lookup is platform-dependent.
+    For example backslashes in source unit name can be interpreted as directory separators or not
+    and the lookup can be case-sensitive or not, depending on the underlying platform.
+
+    For portability it is recommended to avoid using import paths that will work correctly only
+    with a specific import callback or only on one platform.
+
+.. index:: ! CLI path
+
+Populating the Virtual Filesystem
+---------------------------------
+
+The initial content of the VFS depends on how you invoke the compiler:
+
+#. **CLI**
+
+   When you compile a file using the command-line interface of the compiler, you provide one or
+   more *CLI paths* to files containing Solidity code:
+
+   .. code-block:: bash
+
+       solc contract.sol /usr/local/dapp-bin/token.sol
+
+   The source unit name of a file loaded this way is simply the specified CLI path with
+   platform-specific separators converted to forward slashes.
+
+   .. index:: standard JSON
+
+#. **Standard JSON**
+
+   When using the :ref:`Standard JSON <compiler-api>` API (via either the `JavaScript interface
+   <https://github.com/ethereum/solc-js>`_ or the ``--standard-json`` command-line option)
+   you provide input in JSON format, containing, among other things, the content of all your source
+   files:
+
+   .. code-block:: json
+
+       {
+           "language": "Solidity",
+           "sources": {
+               "contract.sol": {
+                   "content": "import \"./util.sol\";\ncontract C {}"
+               },
+               "util.sol": {
+                   "content": "library Util {}"
+               },
+               "/usr/local/dapp-bin/token.sol": {
+                   "content": "contract Token {}"
+               }
+           },
+           "settings": {"outputSelection": {"*": { "*": ["metadata", "evm.bytecode"]}}}
+       }
+
+   The ``sources`` dictionary becomes the initial content of the virtual filesystem and its keys
+   are used as source unit names.
+
+#. **Standard JSON (via import callback)**
+
+   With Standard JSON it is also possible to tell the compiler to use the import callback to obtain
+   the source code:
+
+   .. code-block:: json
+
+       {
+           "language": "Solidity",
+           "sources": {
+               "/usr/local/dapp-bin/token.sol": {
+                   "urls": [
+                       "/projects/mytoken.sol",
+                       "https://example.com/projects/mytoken.sol"
+                   ]
+               }
+           },
+           "settings": {"outputSelection": {"*": { "*": ["metadata", "evm.bytecode"]}}}
+       }
+
+   If an import callback is available, the compiler will pass it the source unit names specified in
```

Comment:
the compiler will pass ~it~ the source unit names specified

Reply:
`it` here means the callback. Maybe it would be clearer as
```
If an import callback is available, the compiler will give it the source unit names specified in
```
?

Reply:
Replaced with `give`.